packages/backend/src/payment-method/ilp/spsp/middleware.ts

```diff
@@ -24,11 +37,20 @@ const spspMiddleware = async (
 ): Promise<void> => {
   // Fall back to legacy protocols if client doesn't support Open Payments.
   if (ctx.accepts('application/spsp4+json')) {
-    const receiver = ctx.walletAddress ?? ctx.incomingPayment
-    ctx.paymentTag = receiver.id
+    const walletAddressService = await ctx.container.use('walletAddressService')
+
+    const walletAddress = await walletAddressService.getByUrl(
+      ctx.walletAddressUrl
```

Comment:
Using `ctx.walletAddressUrl` instead of `ctx.walletAddress` as part of a larger refactor to remove the wallet address middleware altogether